### PR TITLE
Task/bugfix refactor develop

### PR DESCRIPTION
**Pull request by smyaseen**
## GOAL & WORK

1. Added Image URL feature
2. Resolved every warnings
3. Fixed bug on form validations
4. Refactor and bugfixes

### CODE REVIEW

File: src/Features/Admin/Categories/AddCategory/index.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const AddCategory = () => {
     const { validateArray, isValid } = validateOnSubmit(fields, true);
     setFields(validateArray);
     if (isValid) {
-      const name = fields.map(({ value }, index) => value);
+      const name = fields.map(({ value }) => value);
```
Comment (SiddiquiMonis401) [inline src/Features/Admin/Categories/AddCategory/index.js:43]: when we are giving name[0] below why do we need to do the map over fileds array ??

File: src/Features/Admin/Users/EditUser/index.js [hunk 2]
```diff
@@ -156,7 +157,7 @@ const EditUser = () => {
   const saveHandler = () => {
     const { validateArray, isValid } = validateOnSubmit(fields, true);
     setFields(validateArray);
-
+    // console.log(isValid, fields);
```
Comment (SiddiquiMonis401) [inline src/Features/Admin/Users/EditUser/index.js:160]: remove console

File: src/Features/Vendor/BalanceSheet/mutations.js [hunk 3]
```diff
@@ -37,10 +37,10 @@ export const DeleteBalanceById = () => {
     },
 
     {
-      onSuccess: (data, variables, context) => {
+      onSuccess: () => {
         // success!
       },
-      onError: (error, variables, context) => {
+      onError: () => {
```
Comment (SiddiquiMonis401) [inline src/Features/Vendor/BalanceSheet/mutations.js:43]: When there is not any action that needs to be performed on success and error just remove both !!

File: src/Features/Vendor/Menu/Add Menu/index.js [hunk 4]
```diff
@@ -7,14 +7,14 @@ import { useHistory } from 'react-router';
...
-import { ERROR, GetHeader } from '../../../../scripts/constants';
+import { ERROR, GetHeader, imgURLRegex } from '../../../../scripts/constants';
 import { validateOnSubmit, SelectChangeHandler, fieldChangeHandler } from '../../../../util/CommonGridBasedFormUtils';
 import { logout } from '../../../Auth/actions';
 import { items } from '../../mutation';
 import { FetchCategories, FetchRestaurants } from '../../request';
 const AddMenu = () => {
   const { headers } = GetHeader();
-  const [isSubmit, setSubmit] = useState(false);
+  const [, setSubmit] = useState(false);
```
Comment (SiddiquiMonis401) [inline src/Features/Vendor/Menu/Add Menu/index.js:17]: Why we need the state setter when we did not use the state ??

**Review by SiddiquiMonis401 — CHANGES_REQUESTED** [review]

**Review by ShahirZain — APPROVED** [review]

**Pull request merged**